### DISCUSSION TITLE
Try to add a new supported device (PIR sensor)

### DISCUSSION THREAD

**Discussion by MaximeNagel** [make-all/tuya-local#1025] [General]
Hi everyone,

I have try to add a new supported device creating a new yaml file.
But I see that I have a parse error when checking HA log when I try to add a new device.

Can you please help me to figure out what's the problem? I already try to validate the file using a yaml online tool, and it says that yaml is correct.

[zg205w_mmWave_presence_sensor.txt](https://github.com/make-all/tuya-local/files/12436986/zg205w_mmWave_presence_sensor.txt)

Error in HA (but it doesn't give any useful information)
Logger: custom_components.tuya_local.helpers.device_config
Source: custom_components/tuya_local/helpers/device_config.py:950
Integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: 24 août 2023 à 19:43:34 (6 occurrences)
Last logged: 09:48:37

Parse error in zg205w_mmWave_presence_sensor.yaml

**Comment by andbuitra** [comment]
I may be wrong but the first secondary entity (the light entity) doesn't have a name. Perhaps that's it?

**Reply by MaximeNagel** [reply]
Thanks for your help, I have made the fix and a few other adjustments, but still the same issue

File : 
`name: ZG-205W mmWave presence sensor
products:
  - id: qbapy7r4deyp6zbx
    name: ZG-205W
primary_entity:
  entity: binary_sensor
  class: occupancy
  dps:
    - id: 1
      type: enum
      name: sensor
      mapping:
        - dps_val: presence
          value: true
        - dps_val: none
          value: false
secondary_entities:
  - entity: switch
    name: Led Indicator
    category: config
    dps:
      - id: 103
        type: boolean
        name: switch
  - entity: number
    name: Sensitivity
    category: config
    icon: mdi:motion-sensor
    dps:
      - id: 2
        type: integer
        name: value
        range:
          min: 0
          max: 10
  - entity: number
    name: Stationary sensitivity
    category: config
    icon: mdi:motion-sensor
    dps:
      - id: 102
        type: integer
        name: value
        range:
          min: 0
          max: 10
  - entity: number
    name: Small motion detection sensitivity
    category: config
    icon: mdi:motion-sensor
    dps:
      - id: 106
        type: integer
        name: value
        range:
          min: 0
          max: 10
  - entity: number
    name: Motion detection distance
    category: config
    icon: mdi:signal-distance-variant
    dps:
      - id: 3
        type: integer
        name: value
        unit: m
        range:
          min: 0
          max: 1000
        mapping:
          - scale: 100
            step: 10
  - entity: number
    name: Stationary detection distance
    category: config
    icon: mdi:signal-distance-variant
    dps:
      - id: 4
        type: integer
        name: value
        unit: m
        range:
          min: 0
          max: 600
        mapping:
          - scale: 100
            step: 10
  - entity: number
    name: Small motion detection distance
    category: config
    icon: mdi:signal-distance-variant
    dps:
      - id: 105
        type: integer
        name: value
        unit: m
        range:
          min: 0
          max: 600
        mapping:
          - scale: 100
            step: 10
  - entity: sensor
    name: Alarm Volume
    class: enum
    icon: mdi:list-status
    category: config
    dps:
      - id: 107
        type: enum
        name: sensor
        mapping:
          - dps_val: high
            value: High
          - dps_val: middle
            value: Middle
          - dps_val: low
            value: Low
          - dps_val: mute
            value: Mute
  - entity: sensor
    name: Alarm Mode
    class: enum
    icon: mdi:list-status
    category: config
    dps:
      - id: 104
        type: enum
        name: sensor
        mapping:
          - dps_val: armed
            value: Armed
          - dps_val: disarm
            value: Disarmed
          - dps_val: alarm
            value: Alarm
  - entity: sensor
    name: Detection state
    class: enum
    icon: mdi:list-status
    dps:
      - id: 116
        type: enum
        name: sensor
        mapping:
          - dps_val: Large move
            value: Large move
          - dps_val: Small move
            value: Small move
          - dps_val: Breathe
            value: Breathe
  - entity: number
    name: Presence Keep Time
    category: config
    icon: mdi:timer-sand
    dps:
      - id: 108
        type: integer
        name: value
        unit: s
        range:
          min: 1
          max: 100
        mapping:
          - scale: 10
            step: 1
  - entity: number
    name: Alarm Time
    category: config
    icon: mdi:timer
    dps:
      - id: 109
        type: integer
        name: value
        unit: s
        range:
          min: 1
          max: 600
        mapping:
          - scale: 1
            step: 10
  - entity: sensor
    class: illuminance
    dps:
      - id: 101
        type: integer
        name: sensor
        unit: lx
        class: measurement
`

**Comment by make-all** [comment]
enum is not a recognised type.  It should be string.

**Reply by MaximeNagel** [reply]
Such a noob mistake... Thx for your help! Much appreciate. 
It seems to work pretty well now : 
![image](https://github.com/make-all/tuya-local/assets/6407540/b9afd938-e3ab-4215-af93-59eadc5ff382)

- [X ] Maybe just another improvement. Can I make the alarm mode and alarm volume à select input to change the value and not only display it's current state ?  : Edited I found it myself...
- I'm going to make few changes , mostly icon and I think everything will be done.

Do you want me to add the new sensor myself(with a merge request) or I'm sending you the yaml file (with all correction) ?

And do you want me to make any other adjustement? I have a try to respect the doc file but maybe I missed something!

**Comment by make-all** [comment]
Whether you can make the Alarm mode and volume editable depends on the device.  If the Query Things Data Model API call shows them as rw (read-write) then probably it is possible.  You could also look at the alarm_control_panel and siren entity types to see if they are suitable.

**Reply by MaximeNagel** [reply]
Yep, I read again the doc you provide and use the select entity type. It's much better.

![image](https://github.com/make-all/tuya-local/assets/6407540/52d6f8da-84c3-4a3b-b296-c687b8dd4cda)

Any other recommendations before I submit the PR ? 

I still need to tweak some icon I think
